base_layer/wallet/src/transaction_service/multisig/session.rs

```diff
@@ -0,0 +1,552 @@
+        let mut message_bytes = [0u8; 32];
+        message_bytes.clone_from_slice(challenge_key.pub_key.as_bytes());
+        let message: Box<[u8; 32]> = Box::new(message_bytes);
+
+        let tx_id = TxId::new_random();
+        let uuid = Uuid::new_v4();
+        let user_data = uuid.as_bytes().to_vec();
+        let fee_per_gram = MicroMinotari::from(1);
+        let payment_id =
+            MemoField::new_address_and_data(recipient.clone(), fee_per_gram, true, TxType::PaymentToOther, user_data)
+                .map_err(|e| TransactionServiceError::Other(format!("Failed to create MemoField: {}", e)))?;
+
+        let mut selected_criteria = UtxoSelectionCriteria::default();
+
+        selected_criteria.excluding_multisig = true;
+
+        let mut stp = output_manager
+            .prepare_transaction_to_send(
+                tx_id,
+                amount,
+                selected_criteria,
+                OutputFeatures::default(),
+                fee_per_gram,
+                push_pubkey_script(&Default::default()),
+                Covenant::default(),
+            )
+            .await?;
+
+        let sender_offset_key = self
+            .resources
+            .transaction_key_manager_service
+            .get_next_key(TransactionKeyManagerBranch::OneSidedSenderOffset.get_branch_key())
+            .await?;
+
+        let recipient_spend_key = recipient.public_spend_key();
+
+        let sender_offset_public_key = sender_offset_key.pub_key;
+
+        let encrypted_data_shared_secret = key_manager
+            .get_diffie_hellman_shared_secret(
+                &sender_offset_key.key_id,
+                recipient
+                    .public_view_key()
+                    .ok_or(TransactionServiceError::Other("Missing public view key".to_string()))?,
+            )
+            .await?;
+
+        let encryption_private_key = shared_secret_to_output_encryption_key(&encrypted_data_shared_secret)
+            .map_err(|e| TransactionServiceError::Other(format!("Failed to derive encryption key: {}", e)))?;
+
+        let encryption_key_id = key_manager.import_key(encryption_private_key.clone()).await?;
+
+        let mut ephemeral_pubkeys = Vec::new();
+        for pk in &public_keys {
+            // 1. Get the shared secret (aD)
+            let shared_secret = key_manager
+                .get_diffie_hellman_shared_secret(&sender_offset_key.key_id, &pk)
+                .await?;
+
+            // 2. Hash the shared secret for stealth domain separation
+            let stealth_hash = diffie_hellman_stealth_domain_hasher(shared_secret);
+
+            let mask_key_id = &key_manager
+                .import_key(PrivateKey::from_uniform_bytes(stealth_hash.as_ref())?)
+                .await?;
+
+            let ephemeral_pubkey = key_manager
+                .clone()
+                .stealth_address_script_spending_key(&mask_key_id, pk)
+                .await?;
+
+            ephemeral_pubkeys.push(ephemeral_pubkey.clone());
```

Comment:
```suggestion
            let shared_secret = key_manager
                .get_diffie_hellman_shared_secret(&sender_offset_key.key_id, &pk)
                .await?;
               
               let stealth =  shared_secret.to_public_key();
               let ephemeral_pubkey = pk + stealth;
               ephemeral_pubkeys.push(ephemeral_pubkey);
```
This is way different from the original scheme, and does a ton of extra stuff thats not needed